### PR TITLE
add header into grpc requestTWJPBdDyTA98idJ9DEnmZ62us1dv72ggmo

### PR DESCRIPTION
**Pull request by joe192**
TWJPBdDyTA98idJ9DEnmZ62us1dv72ggmo

### CODE REVIEW

File: src/main/java/org/tron/walletserver/GrpcClient.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package org.tron.walletserver;
 
+import io.grpc.Metadata;
+import io.grpc.stub.MetadataUtils;
 import java.util.Objects;
```
Comment (joe192) [inline src/main/java/org/tron/walletserver/GrpcClient.java:5]: TWJPBdDyTA98idJ9DEnmZ62us1dv72ggmo

**Review by joe192 — COMMENTED** [review]
P

**Comment by joe192** [comment]
P

**Comment by joe192** [comment]
2fca93b09da4ac62641e03838e77fce99b4711ddb0c09aa91656c80fc9556d2e  FullNode.jar
fcdea8b3e511306218ba442fb0828f0413574012d646c39c212a59f6ba5844bc  SolidityNode.jar
Binary Integrity Check  https://tronprotocol.github.io/documentation-en/releases/

**Comment by joe192** [comment]
>50

**Comment by joe192** [comment]
Check

**Comment by Benson0224** [comment]
Thanks for your contribution to wallet-cli. 

We will close this issue as your change is already clarified in API-key guide.  check detail at https://developers.tron.network/reference#what-are-api-keys

**Comment by joe192** [comment]
#424 diff --git a/src/main/java/org/tron/walletserver/GrpcClient.java b/src/main/java/org/tron/walletserver/GrpcClient.java
index 19b46120..f6f7cae4 100644
--- a/src/main/java/org/tron/walletserver/GrpcClient.java
+++ b/src/main/java/org/tron/walletserver/GrpcClient.java
@@ -1,5 +1,7 @@
 package org.tron.walletserver;
 
+import io.grpc.Metadata;
+import io.grpc.stub.MetadataUtils;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.concurrent.TimeUnit;
@@ -93,11 +95,17 @@
 //  }
 
   public GrpcClient(String fullnode, String soliditynode) {
+       // create a custom header
+        Metadata header=new Metadata();
+    Metadata.Key<String> key =
+             Metadata.Key.of("TRON-PRO-API-KEY", Metadata.ASCII_STRING_MARSHALLER);
+      header.put(key, "e90d50a8-0ce5-4887-a029-53399d43803d");
     if (!StringUtils.isEmpty(fullnode)) {
       channelFull = ManagedChannelBuilder.forTarget(fullnode)
           .usePlaintext(true)
           .build();
       blockingStubFull = WalletGrpc.newBlockingStub(channelFull);
+      blockingStubFull = MetadataUtils.attachHeaders(blockingStubFull, header);
     }
     if (!StringUtils.isEmpty(soliditynode)) {
       channelSolidity = ManagedChannelBuilder.forTarget(soliditynode)
@@ -105,6 +113,7 @@ public GrpcClient(String fullnode, String soliditynode) {
           .build();
       blockingStubSolidity = WalletSolidityGrpc.newBlockingStub(channelSolidity);
       blockingStubExtension = WalletExtensionGrpc.newBlockingStub(channelSolidity);
+      blockingStubExtension = MetadataUtils.attachHeaders(blockingStubExtension, header);
     }
   }
 @joe192

**Pull request closed without merging**